### DISCUSSION TITLE
How to run custom function on realtime Presence leave event?

### DISCUSSION THREAD

**Discussion by karandwivedi42** [supabase/supabase#13501] [Questions]
I want to use Supabase Presence to connect multiple iot devices to a central server and keep track of uptime and connected devices.

Based on reading the docs, there is no server side logging of connection / disconnection event. To achieve this, I can keep a small service running on fly.io which runs a supabase client that listens to presence events and logs that to supabase database.

However, it would be great if there was a way to do this within supabase. For example -

1. If there is a known supabase postgres table that keeps a track of connected devices, where I can add triggers for CREATE / DELETE.
2. If we can attach a supabase edge function to run when presence changes.

Am I missing something here or is the method I mentioned the only way to keep a log of presence?

**Comment by fardeem** [comment]
@karandwivedi42 did you end up solving this?

**Reply by GaryAustin1** [reply]
Realtime only has info on the Postgres server for postgres changes.  Broadcast and Presence are both handled in the realtime server without database connection.

**Reply by fardeem** [reply]
ahh does this mean for realtime there is no native way to get a callback when a user leaves?

**Reply by GaryAustin1** [reply]
I don't use presence or broadcast myself, but they don't involve the database at all.  The client appears to get notice when a user leaves, https://supabase.com/docs/guides/realtime/presence#sync-and-track-state
